packages/cli/config/jest.js

```diff
@@ -131,10 +131,38 @@ async function getProjectConfig(targetPath, displayName) {
     },
 
     transform: {
-      '\\.(js|jsx|ts|tsx|mjs|cjs)$': [
+      '\\.(mjs|cjs)$': [
         require.resolve('./jestSwcTransform'),
         {
           sourceMaps: envOptions.enableSourceMaps || envOptions.nextTests,
+          jsc: {
+            target: 'es2019',
+            parser: {
+              syntax: 'ecmascript',
+            },
+            transform: {
+              react: {
+                runtime: 'automatic',
+              },
+            },
+          },
+        },
+      ],
+      '\\.(t|j)sx?$': [
+        require.resolve('./jestSwcTransform'),
+        {
+          sourceMaps: envOptions.enableSourceMaps || envOptions.nextTests,
+          jsc: {
+            parser: {
+              syntax: 'typescript',
+              tsx: true,
```

Comment:
realized we can't have `tsx` enabled for regular `.ts` files as well. There's syntax you can use in `.ts` but not `.tsx`, like `<string> foo`

Reply:
Something like this?

Reply:
yep!